CHANGES.md

```diff
@@ -1,6 +1,427 @@
+* Fixed a problem with combinations of `const ref`, `owned`, and `coforall`
+* Fixed a bug that caused sporadic problems  with the `--memLog` flag
+* added a remote cache fence to barrier calls
+* fixed a bug with overload sets and operators
+
+
+Bug Fixes for Build Issues
+--------------------------
+* enabled building with GPU support for host compilers other than clang
+
+Bug Fixes for Libraries
+-----------------------
+* fixed bug with `bigint.removeFactor()` when `fac` is `0`
+
+Bug Fixes for Tools
+-------------------
+* fixed the naming of `CHPL_` variables in the output of `chpl --help-env`
+
+Platform-specific Bug Fixes
+---------------------------
+
+Third-Party Software Changes
+----------------------------
+* updated the bundled LLVM and clang to version 13
+* updated the bundled version of `libfabric` to version 1.13.2
+* updated the Sphinx version used for `chpldoc`
+* updated the Sphinx Chapel domain version used for `chpldoc`
+* updated the breathe version used for `chpldoc`
+
+Developer-oriented changes: Process
+-----------------------------------
+* trailing whitespace is now flagged in CI using GitHub actions
+
+Developer-oriented changes: Documentation
+-----------------------------------------
+
+Developer-oriented changes: Naming Changes
+------------------------------------------
+* coined the name 'dyno' to refer to aspects of the compiler rewrite effort
+
+Developer-oriented changes: Module changes
+------------------------------------------
+* simplified the barrier used to initialize locales
+* updated 'Socket' library operators to use the `operator` keyword
+* added `iostyleInternal` type for use when `iostyle` would have been needed
+
+Developer-oriented changes: Performance improvements
+----------------------------------------------------
+
+Developer-oriented changes: Makefile / Build-time changes
+---------------------------------------------------------
+* adjusted unique build path for `hwloc` and `re2`
+* stopped inappropriately building the bundled GMP when `CHPL_GMP=system`
+* improved ordering of `-I`/`-L` arguments for bundled vs. system versions
+* non-debug builds of `chpl` now use `-DNDEBUG` to disable asserts in C++ code
+* removed the `make third-party` target
+* a 'dyno'-specific C++ linter now checks some class methods
+
+Developer-oriented changes: Compiler Flags
+------------------------------------------
+* added support for a `-j`/`--parallel-make` flag for the C back-end
+* renamed the `--compiler-library-parser` flag to `--dyno`
+* added `--debug-trace` & `--break-on-hash` flags for debugging 'dyno's parser
+
+Developer-oriented changes: Compiler improvements/changes
+---------------------------------------------------------
+* updated `--incremental` to compile each module's `.c` file indepdently
+* enabled support for `make -j` using the C back-end via `-j` / `--incremental`
+
+Developer-oriented changes: 'dyno' Compiler improvements/changes
```

Comment:
We will need a link to something that explains the dyno name.

Reply:
The first use of the term in the file is:

```
* coined the name 'dyno' to refer to aspects of the compiler rewrite effort
```

which I think makes it fairly self-explanatory in the context of this file (esp. given that all mentions are in the "Developer-oriented" sections).